various documentation updates

nashif opened the pull request.

nashif committed:
doc: remove links to wiki

Wiki is being obsoleted, so remove any links that might become dead
really soon.

Signed-off-by: Anas Nashif <anas.nashif@intel.com>

nashif committed:
doc: update macOS getting started documentation

Fixed documentation and updated config files for xtools to be used with
the latest version of crosstool-ng (1.23)

Jira: ZEP-616, ZEP-2146
Signed-off-by: Anas Nashif <anas.nashif@intel.com>

nashif committed:
xtools: add new configurations for xtools 1.23

Remove arm.config, we should be using the official ARM cross compiler
instead.

Signed-off-by: Anas Nashif <anas.nashif@intel.com>

nashif committed:
doc: also require dtc to be installed for linux

Signed-off-by: Anas Nashif <anas.nashif@intel.com>

carlescufi commented on doc/getting_started/installation_mac.rst:
```diff
@@ -9,6 +9,7 @@ After completing these steps, you will be able to compile and run your Zephyr
 applications on the following Mac OS version:
 
 * Mac OS X 10.11 (El Capitan)
+* maxOS Sierra 10.12
```
s/maxOS/macOS. Although I do like the idea of a "Max OS" :)

carlescufi commented on doc/getting_started/installation_mac.rst:
```diff
@@ -55,10 +55,17 @@ the brew command line.
 .. code-block:: console
 
-   $ brew install gettext qemu help2man mpfr gmp coreutils wget python3 dfu-util
+   $ brew install gettext qemu help2man mpfr gmp coreutils wget python3 dfu-util dtc
```
What I feel is missing here is a split between tools and toolchain. For ARM, it is much easier to use GNU ARM Embedded and only install qemu and dtc, so ideally it would be great if we could separate those in 2 different steps

nashif committed:
doc: emphasize usage of MSYS2 MSYS Shell

Some users started the wrong shell (MinGW) and ended up having build
issues, added a note about starting the right shell.

Jira: ZEP-2004
Signed-off-by: Anas Nashif <anas.nashif@intel.com>

dbkinder commented on boards/boards.rst:
```diff
@@ -3,6 +3,8 @@
 Supported Boards
 ################
 
+To add a new board, please use the template available under
```
To add support documentation for a new board, ...

dbkinder commented on doc/getting_started/installation_mac.rst:
```diff
@@ -9,6 +9,7 @@ After completing these steps, you will be able to compile and run your Zephyr
 * Mac OS X 10.11 (El Capitan)
+* macOS Sierra 10.12
 
 Developing for Zephyr on OS X generally requires you to build the
```
Looks like the Mac OS is now called "macOS" so the rest of these references to "OS X" should probably be change to "macOS", yes?

carlescufi replied:
Yes, the transition from "OS X" to "macOS" (it was only called Mac OS X years and years ago) happened in 10.12

dbkinder reviewed (changes requested):
a few edits

nashif commented:
updated, ready to go
